### ISSUE TITLE
withXXX and SortedMap always returns empty

### ISSUE THREAD

**Issue by diamondq** [immutables/immutables#419]
Reproducible in 2.2.11:

Using a plain J2SE (ie. no Guava), the implementation of a withXXX and a SortedMap uses a helper method called createUnmodifiableSortedMap. Unfortunately, the one path that checks nulls or skips nulls uses the result map (sortedMap) instead of the input map (map) for the for loop. Thus, it always ends up with an empty map.

```java
   TreeMap<K, V> sortedMap = reverse
        ? new TreeMap<K, V>(Collections.reverseOrder())
        : new TreeMap<K, V>();

    if (checkNulls || skipNulls) {
      for (Map.Entry<K, V> e : sortedMap.entrySet()) {
        K k = e.getKey();
        V v = e.getValue();
```

A simple change to

```java
      for (Map.Entry<K, V> e : map.entrySet()) {
```

would fix the problem.

**Comment by elucash** [comment]
Unfortunate omission, thank you for revealing this!

**Comment by elucash** [comment]
released in 2.2.12
